src/objects.h

```diff
@@ -716,10 +850,30 @@ class Stack : public HeapObject {
   }
 
  private:
+  // We keep a 'guard zone' of words that must not be touched right after
+  // the header of the stack object. If the stack overflows into the guard
+  // zone we will catch the issue when enter or leave the interpreter - or
+  // when we transfer control between tasks.
+#ifdef BUILD_32
+  static const uword GUARD_ZONE_MARKER = 0xcaadabe7;
+#elif BUILD_64
+  static const uword GUARD_ZONE_MARKER = 0x7eb91112caadabe7;
+#endif
+#ifdef DEBUG
+  static const int GUARD_ZONE_WORDS = 8;
+#else
+  // TODO(kasper): We do not want to pay for the guard zone in deployments,
+  // so we should keep the zone empty there after a bit of testing..
+  static const int GUARD_ZONE_WORDS = 4;
+#endif
+  static const int GUARD_ZONE_SIZE = GUARD_ZONE_WORDS * WORD_SIZE;
+
   static const int LENGTH_OFFSET = HeapObject::SIZE + WORD_SIZE;
   static const int TOP_OFFSET = LENGTH_OFFSET + WORD_SIZE;
   static const int TRY_TOP_OFFSET = TOP_OFFSET + WORD_SIZE;
-  static const int HEADER_SIZE = TRY_TOP_OFFSET + WORD_SIZE;
+  static const int PENDING_STACK_CHECK_METHOD_OFFSET = TRY_TOP_OFFSET + WORD_SIZE;
+  static const int GUARD_ZONE_OFFSET = PENDING_STACK_CHECK_METHOD_OFFSET + WORD_SIZE;
```

Comment:
Add a comment that the stack grows down.

Reply:
Done.